### PR TITLE
Material crud

### PR DESCRIPTION
**Pull request by mdaudiffret**
- migration AddAdminToUsers
- migration AddCategoryToMaterial
- Enabled index for materials only to admin
- Enabled new for materials only to admin

### CODE REVIEW

File: app/controllers/materials_controller.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  end
+
+  def edit
+    @material = Material.find(params[:id])
+    authorize @material
+  end
+
+  def update
+    @material = Material.find(params[:id])
+    @material.update(material_params)
```
Comment (Nicode80) [inline app/controllers/materials_controller.rb:34]: Maybe we could use a if statement and render edit if failed like for the save ?

File: app/controllers/materials_controller.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    authorize @material
+    redirect_to material_path(@material)
+  end
+
+  def destroy
+    @material = Material.find(params[:id])
+    @material.destroy
+    authorize @material
```
Comment (Nicode80) [inline app/controllers/materials_controller.rb:43]: What is the interets to authorize this @ material if destroyed ?

File: app/views/materials/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<div class="container">
+  <div class="d-flex align-items-center justify-content-between">
+    <h1>Materials</h1>
+    <%= link_to 'Add material', new_material_path, class: "btn btn-primary" %>
+  </div>
+  <%= simple_form_for(@material) do |f| %>
+    <%= f.input :name %>
+    <%= f.input :description %>
+    <%= f.input :category, as: :select, collection: Material::CATEGORIES %>
+    <%= f.input :phoot, as: :file %>
```
Comment (Nicode80) [inline app/views/materials/new.html.erb]: typo photo and no phoot

File: app/controllers/materials_controller.rb (hunk 2 above)
Comment (AshFromAvalon) [reply app/controllers/materials_controller.rb:43]: @Nicode80 the interest is that you neeed to authorize the variable for Pundit to check the policy. Pundit need to check the following condition: `record.user == user` (in Pundit user = current_user and record is, in this case `@material`, that you need to authorize for Pundit to access it). The convention is to place authorize at the bottom of your method.

File: app/views/materials/index.html.erb [hunk 4]
```diff
@@ -7,7 +7,7 @@
     <% @materials.each do |material| %>
       <li class="list-group-item d-flex justify-content-between align-items-center">
         <div class="d-flex align-items-center justify-content-between">
-          <%= cl_image_tag, height: 20 , width: 20, crop: :fill %>
+          <%#= cl_image_tag, mheight: 20 , width: 20, crop: :fill %>
```
Comment (AshFromAvalon) [inline app/views/materials/index.html.erb]: I think you need the attachement --> `cl_image_tag, **metarial.photo.key**` for it to work

**Review by AshFromAvalon — COMMENTED** [review]
@Max, I think miss an argument in cl_image_tag + see Nico's comment for the typo in the form

File: app/controllers/materials_controller.rb (hunk 1 above)
Comment (mdaudiffret) [reply app/controllers/materials_controller.rb:34]: Edit not finished yet

File: app/views/materials/new.html.erb (hunk 3 above)
Comment (mdaudiffret) [reply app/views/materials/new.html.erb]: thks

File: app/views/materials/index.html.erb (hunk 4 above)
Comment (mdaudiffret) [reply app/views/materials/index.html.erb]: see next commit

File: app/controllers/instructions_controller.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
+class InstructionsController < ApplicationController
+  def new
+    @material = Material.find(params[:material_id])
+    @instruction = Instruction.new
+    authorize @instruction
+  end
+
+  def create
+    @instruction = Instruction.new(instruction_params)
+    @material = Material.find(params[:material_id])
```
Comment (Nicode80) [inline app/controllers/instructions_controller.rb:10]: Maybe you can refacto this one and put in a before action for new, create and edit ?

**Review by Nicode80 — APPROVED** [review]

**Pull request merged**